src/libraries/System.Memory/src/System/Buffers/SequenceReader.Search.cs

```diff
@@ -75,7 +75,7 @@ private bool TryReadToSlow(out ReadOnlySpan<T> span, T delimiter, T delimiterEsc
                 return false;
             }
 
-            Debug.Assert(sequence.Length > 0);
+            Debug.Assert(!sequence.IsEmpty);
```

Comment:
@stephentoub could `ReadOnlySequence<T>` be another type that we may need to exclude?